### ISSUE TITLE
Importing puppeteer as ES module from TypeScript not working

### ISSUE THREAD

**Issue by dd-pardal** [puppeteer/puppeteer#7193]
### Steps to reproduce

* Puppeteer version: 9.0.0
* Platform / OS version: Windows 10 (64-bit)
* URLs (if applicable):
* Node.js version: 15.5.1

**What steps will reproduce the problem?**

1. Create a new directory and `cd` into it.
2. Run `npm install puppeteer-core`.
3. Add `"type": "module"` to `package.json`.
4. Create a `tsconfig.json` with the following content:
   ```json
   {
       "compilerOptions": {
           "target": "ES2020",
           "module": "ESNext",
           "lib": ["ES2020", "DOM"],
           "moduleResolution": "node",
           "outDir": "build",
           "strict": true
       },
       "include": ["**/*"]
   }
   ```
5. Create a `index.ts` file with the following content:
   ```ts
   import puppeteer from "puppeteer-core";
   console.log(puppeteer.connect);
   ```
6. Run `tsc`.

**What is the expected result?**

The project compiles and when `node build/index.js` is executed, `[Function: connect]` is logged into the console.

**What happens instead?**

The project fails to compile.

**More information**

The following code compiles, but logs `undefined`. The only property in `puppeteer` is named `default`.
```ts
import * as puppeteer from "puppeteer-core";
console.log(puppeteer.connect);
```

Basically TypeScript thinks the correct thing to do is `import * as puppeteer` while Node.js only accepts `import puppeteer` (default import).

**Comment by dd-pardal** [comment]
This error can be fixed by setting the `allowSyntheticDefaultImports` compiler option to `true`.

**Comment by impetusdev** [comment]
> 

Could you explain the reasoning?

**Comment by dd-pardal** [comment]
> > 
> 
> Could you explain the reasoning?

Not sure what you mean by that. I thought this was a problem with Puppeteer but apparently the problem was on my side. I needed to set `allowSyntheticDefaultImports` to `true` because of the way Puppeteer exports things.
